File: tests/fixtures/realworld/dart/easy/calculator.dart
/// A simple calculator.
class Calculator {
  int result = 0;

  Calculator(this.result);

  /// Add a value to the result.
  int add(int value) => result += value;

  int subtract(int value) => result -= value;

  static String version() => '1.0.0';
}

/// User account model.
class User {
  final int id;
  final String name;

  User(this.id, this.name);

  String greet() => 'Hi $name';
}

/// Application status.
enum Status { active, inactive, banned }

/// Top-level helper function.
int doubleIt(int x) => x * 2;

File: tests/fixtures/realworld/scala/easy/Calculator.scala
/** A simple calculator. */
class Calculator {
  private var result: Int = 0

  def add(value: Int): Int = {
    result += value
    result
  }

  def getResult: Int = result
}

/** User account model. */
case class User(id: Int, name: String)

/** Service logger interface. */
trait Logger {
  def log(message: String): Unit
}

/** Application status. */
enum Status {
  case Active, Inactive, Banned
}

/** Top-level helper. */
def doubleIt(x: Int): Int = x * 2
